Internal: Documentation of `document` component. [ED-5444] (#16810)
elementor/elementor: docs/assets/dev/js/editor/document/readme.md

# Legend
* Example:
  ![legend-section-widget-tabs](_images/legend-section-widget-tabs.png)
- ![#ffd29b](https://via.placeholder.com/15/ffd29b/000000?text=+)  `Section ID` - `98c3441`
- ![#93ff8b](https://via.placeholder.com/15/93ff8b/000000?text=+) `Column ID` &nbsp;&nbsp;- `18f2ab4`, `4b957d9`

> - ![#43f7ff](https://via.placeholder.com/15/43f7ff/000000?text=+)  `Widget ID` &nbsp;&nbsp;- `5412776`

Reviewer: What is this?

Plus add some introduction to the component

## Component -- `$e.components.get('document')`

*  **Name**: Document.

> *  **Description**: Document manipulation.

Reviewer: Explain

> ## All **Document** components

Reviewer: Maybe "*sub*" components?

| Component Name                 | Access Component                         | Description
|--------------------------------|------------------------------------------|-----------------------
| [Dynamic](dynamic/readme.md)   | `$e.components.get('document/dynamic')`  | Dynamic enable, disable and change.

> | [Elements](elements/readme.md) | `$e.components.get('document/elements')` | Elements manipulation. create, edit, remove, etc...

Reviewer: ```suggestion
| [Elements](elements/readme.md) | `$e.components.get('document/elements')` | Elements manipulation (create, edit, remove, etc.).
```

> | [History](history/readme.md)   | `$e.components.get('document/history')`  | Provides a way of recording\manipulate history of the commands.

Reviewer: ```suggestion
| [History](history/readme.md)   | `$e.components.get('document/history')`  | Provides a way to record/manipulate the commands history.
```

> | [Repeater](repeater/readme.md) | `$e.components.get('document/repeater')` | Provides a way communicate with repeater.

Reviewer: ```suggestion
| [Repeater](repeater/readme.md) | `$e.components.get('document/repeater')` | Provides a way to communicate with repeaters.
```

> | [Save](save/readme.md)         | `$e.components.get('document/save')`     | Responsible for saving, publish, draft, of the document.

Reviewer: Not sure about that, correct me if I'm wrong.
```suggestion
| [Save](save/readme.md)         | `$e.components.get('document/save')`     | Responsible for saving the document (either publish/draft).
```

> | [UI](ui/readme.md)             | `$e.components.get('document/ui')`       | Irregular component for handling UI shortcuts.

Reviewer: "Irregular"? Explain

Author: Its Irregular because it's only a wrapper for exist command which are :`copy, delete, duplicate, paste, paste-style` only for keyboard shortcuts. where you want to add it?

> ## Component `document/` -- Utils

Reviewer: Maybe something more like this?

```suggestion
## Document Utils
```

> | Util Name                                     | Access Util                                               | Description

Reviewer: Why is the description here if you're not using it?

|-----------------------------------------------|-----------------------------------------------------------|-----------------------
| [findViewRecursive](#)                        | `$e.components.get('document').utils.findViewRecursive()` |
| [findViewById](#)                             | `$e.components.get('document').utils.findViewById()`      |
| [findContainerById](#)                        | `$e.components.get('document').utils.findContainerById()` |

## _Util_ -- `$e.components.get('document').utils.findViewRecursive()`

> *  **Name**: findViewRecursive.

Reviewer: I think it's somewhat redundant
```suggestion
```

*  **Description**: Find view recursively.
*  **Returns**: `{Array}`

   | Property     | Type                  | Requirement   | Description |
       |---           |---                    |---            |---|
   | _parent_     |
   | _key_        |
   | _value_      |

>    | _multiple_   | `{Boolean}`           | **optional**  | default: `{false}`.

Reviewer: the table is broken

## _Util_ -- `$e.components.get('document').utils.findViewById()`
*  **Name**: findViewById.
*  **Description**: Find view by id.

> *  **Returns**: `{ View | Array.<View> }`

Reviewer: ```suggestion
*  **Returns**: `{ View | View[] }`
```

   | Property     | Type                  | Requirement   | Description |
       |---           |---                    |---            |---|
   | _id_         | {String}              | **required**  | Id of the element.

## _Util_ -- `$e.components.get('document').utils.findContainerById()`
*  **Name**: findContainerById.
*  **Description**: Find container by id.
*  **Returns**: `{ Container | Array.<Container> }`

   | Property     | Type                  | Requirement   | Description |
       |---           |---                    |---            |---|
   | _id_         | {String}              | **required**  | Id of the element.

> ### [Back](../../../../../core/common/assets/js/api/core/commands-methods/getall.md) 

Reviewer: Maybe rename the file?

```suggestion
### [Back](../../../../../core/common/assets/js/api/core/commands-methods/get-all.md) 
```
